src/nrnpython/nrnpy_p2h.cpp

```diff
@@ -163,22 +163,21 @@ static void py2n_component(Object* ob, Symbol* sym, int nindex, int isfunc) {
             Py_INCREF(main_module);
             Py_INCREF(main_namespace);
         }
-        tail = PyRun_String(sym->name, Py_eval_input, main_namespace, main_namespace);
+        tail = nb::steal(PyRun_String(sym->name, Py_eval_input, main_namespace, main_namespace));
     } else {
-        Py_INCREF(head);
+        head.inc_ref();
```

Comment:
This seems very iffy. I need to revisit this again.

Reply:
I removed this `inc_ref`. The borrow ensure that the reference count is positive. The assignment will make sure that both `head` and `tail` own their reference (separately).

I'm not 100% convinced this doesn't change anything, e.g. on the `if(pn->type_ == 0)` path, we didn't INCREF `head`, but we `XDECREF` it at the end. If that condition implies that `pn->po_ == nullptr`, then the change is equivalent. If not then I think that the current version would do an additional DECREF.

Reply:
removed.